modules/gke-hub/main.tf

```diff
@@ -38,7 +38,7 @@ resource "google_gke_hub_membership" "default" {
   provider      = google-beta
   for_each      = var.clusters
   project       = var.project_id
-  membership_id = each.key
+  membership_id = reverse(split("/", each.value))[0] # forces re-enrollment of the cluster in the fleet in case when cluster is recreated
```

Comment:
Is there a possibility for this to break if you have clusters with the same name in different projects?

Perhaps you can use something like `replace(each.value, "/", "-")` or keep `each.key` as part of the membership id.

Reply:
I've tested this again and it looks like my workaround was needed only when I did not configure workload identity endpoints. Once they are configured, then hub membership attributes are anyway forcing replacement, so there is no need for change here.